packages/@sanity/schema/test/validation/validation.test.ts

```diff
@@ -171,4 +172,20 @@ describe('Validation test', () => {
       helpId: 'schema-array-of-type-builtin-type-conflict',
     })
   })
+
+  test('accepts blocks with custom style icons', () => {
+    const schemaDef = [
+      {
+        name: 'testBlock',
+        type: 'block',
+        styles: [{icon: SquareIcon, title: 'Normal text', value: 'normal'}],
```

Comment:
Thanks for adding a test! I don't see that we have any tests for passing in a `styles` array without `icon`. Do you think it's worth adding a test for that scenario as well just to make sure that we didn't break anything here by adding `icon`?

Reply:
Added!

Reply:
🙏